src/buffer.rs

```diff
@@ -109,15 +107,6 @@ impl Write for Buffer {
     fn flush(&mut self) -> io::Result<()> {
         self.inner_mut().flush()
     }
-
-    #[inline]
-    fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
-        match self {
-            Buffer::File(file) => file.write_all(buf),
-            Buffer::Stdout(stream) | Buffer::Stderr(stream) => stream.write_all(buf),
-            Buffer::Redirect(stream) => stream.write_all(buf),
-        }
-    }
```

Comment:
Is this because none of these buffer types have an efficient implementation of `write_all()`?

Reply:
My thinking when I wrote it was that if `write_all` has to do many writes then it would only have to do the `match` once, instead of doing it for every individual `write`. But writes are expensive, an extra `match` doesn't matter.

I hadn't thought of efficient `write_all()` implementations, but that's a great point, because there could've been something like that. Writes have to acquire a lock and it's nice if that doesn't have to happen for every individual write. But I benchmarked it and it turns out not to make a difference.

Here's the patch I used, for posterity:
```diff
diff --git a/src/buffer.rs b/src/buffer.rs
index 3a5c437..e983c8c 100644
--- a/src/buffer.rs
+++ b/src/buffer.rs
@@ -114,8 +114,8 @@ impl Write for Buffer {
     fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
         match self {
             Buffer::File(file) => file.write_all(buf),
-            Buffer::Stdout(stream) | Buffer::Stderr(stream) => stream.write_all(buf),
-            Buffer::Redirect(stream) => stream.write_all(buf),
+            Buffer::Stdout(stream) | Buffer::Stderr(stream) => stream.lock().write_all(buf),
+            Buffer::Redirect(stream) => stream.get_mut().lock().write_all(buf),
         }
     }
 }
```